### ISSUE TITLE
Component in library: Error: No value accessor for form control

### ISSUE THREAD

**Issue by supermaz** [angular/angular#32659]
### Affected Package

 The issue is caused by package @angular/....
@agnular/forms

Using angular 8

### Description

We made a custom component implementing ControlValueAccessor which is not very complex:

```
import { Component, forwardRef } from '@angular/core';
import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
@Component({
  selector: 'carta-chart-select',
  template: `
  <span class="pointer fa-stack fa-1x" (click)="valueClicked()">
  <i class="fas fa-chart-line fa-stack-1x" [style.color]="value ? '#000' : '#eee'"></i>
  <i class="fas fa-slash fa-stack-1x" [hidden]="value"></i>
</span>
  `,
  styles: [`
  :host() {
    position: absolute;
    top: 1px;
    right: 1px;
  }
  `],
  providers: [{
    provide: NG_VALUE_ACCESSOR,
    // useExisting: CartaChartSelectComponent,
    useExisting: forwardRef(() => CartaChartSelectComponent),
    multi: true
  }]
})
export class CartaChartSelectComponent implements ControlValueAccessor {
  fnOnChange: (arg0: boolean) => void;
  value: boolean;
  public CartaChartSelectComponent() {
    console.log("CARTALIB: In constructor");
    this.fnOnChange = null;
  }
  valueClicked(): void {
    console.log("CARTALIB: In value clicked");
    this.value = !this.value;
    this.fnOnChange(this.value);
  }
  writeValue(value) {
    console.log("CARTALIB: In write value");
    this.value = value;
  }
  registerOnChange(fn: (arg0: boolean) => void) {
    console.log("CARTALIB: In register");
    this.fnOnChange = fn;
  }
  registerOnTouched(fn) {
  }
}
```

When this component is within my app project, everything works fine. Now we want to build a second app using some shared functionality like this. So I created a library which should contain these modules. As soon as I use this component from the library, I get the above error message. I tracked it a bit down and I found that indeed valueAccessors is null because in the method selectValueAccessor the parameter valueAccessors is also null.

Using this component in a showcase project within the library is also working.

## 🔬 Minimal Reproduction

 https://stackblitz.com/...

The library is quite simple:

public-api.ts:
```
export * from './lib/carta-angular-utilities.module';
export * from './lib/directives/chart-select-button.directive';
```

carta-angular-utilities.module.ts:
```
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms'; // <-- NgModel lives here
import { CartaChartSelectComponent } from './directives/chart-select-button.directive';
import {CommonModule} from '@angular/common';


@NgModule({
  declarations: [CartaChartSelectComponent],
  imports: [
    CommonModule,
    FormsModule, ReactiveFormsModule
  ],
  exports: [CartaChartSelectComponent]
})
export class CARTAAngularUtilitiesModule { }
```

And of course the component from above.

## 🔥 Exception or Error
<pre><code>

ERROR Error: No value accessor for form control with name: 'enabledChartItems.diameter'
    at _throwError (forms.js:2291)
    at setUpControl (forms.js:2201)
    at NgModel.push../node_modules/@angular/forms/fesm5/forms.js.NgModel._setUpStandalone (forms.js:4953)
    at NgModel.push../node_modules/@angular/forms/fesm5/forms.js.NgModel._setUpControl (forms.js:4940)
    at NgModel.push../node_modules/@angular/forms/fesm5/forms.js.NgModel.ngOnChanges (forms.js:4870)
    at checkAndUpdateDirectiveInline (core.js:20865)
    at checkAndUpdateNodeInline (core.js:29225)
    at checkAndUpdateNode (core.js:29187)
    at debugCheckAndUpdateNode (core.js:29821)
    at debugCheckDirectivesFn (core.js:29781)
</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>

Angular CLI: 8.2.0
Node: 10.15.3
OS: win32 x64
Angular: 8.2.0
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.802.0
@angular-devkit/build-angular     0.802.0
@angular-devkit/build-optimizer   0.802.0
@angular-devkit/build-webpack     0.802.0
@angular-devkit/core              8.2.0
@angular-devkit/schematics        8.2.0
@angular/http                     7.2.15
@ngtools/webpack                  8.2.0
@schematics/angular               8.2.0
@schematics/update                0.802.0
rxjs                              6.5.2
typescript                        3.5.3
webpack                           4.38.0
</code></pre>

**Anything else relevant?**
There is also another guy on stackoverflow which seems to have the same issue without a solution:

https://stackoverflow.com/questions/54150044/implementing-controlvalueaccessor-in-our-external-angular-library

**Comment by guobin211** [comment]
**Initial value of variable**
`
fnOnChange: (arg0: boolean) => void;
  value: boolean = false;
  public CartaChartSelectComponent() {
    console.log("CARTALIB: In constructor");
    this.fnOnChange = null;
  }

`

**Comment by supermaz** [comment]
Doesn't makes a difference, same error even when changed to

```
fnOnChange: (arg0: boolean) => void = (v) => { console.log("v:" + v); };
  value: boolean = false;
```

From the console logs it's clear that the constructor is not even called.

**Comment by supermaz** [comment]
I switched to other functions from our project in the meantime and got more funny error messages. So it must have been something deeper. I installed by Angular library with npm link and for this project I'm working on windows. And guess what, npm link seems to be heavily broken...

After I made a manuall install (create folder under node_modules and copy the build files into it) everything from my Angular library works out of the box!

So I close this issue but maybe there is something the Angular Team can do about this? Maybe a warning if symlinks are detected during build?
